hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java

```diff
@@ -0,0 +1,412 @@
+      boolean skipPrevKey = false;
+      String seekKey = prevKeyPrefix;
+      if (StringUtils.isNotBlank(prevKeyPrefix)) {
+        skipPrevKey = true;
+        Table.KeyValue<String, RepeatedOmKeyInfo> seekKeyValue =
+            keyIter.seek(seekKey);
+        // check if RocksDB was able to seek correctly to the given key prefix
+        // if not, then return empty result
+        // In case of an empty prevKeyPrefix, all the keys are returned
+        if (seekKeyValue == null ||
+            (StringUtils.isNotBlank(prevKeyPrefix) &&
+                !seekKeyValue.getKey().equals(prevKeyPrefix))) {
+          return deletedKeyAndDirInsightInfo;
+        }
+      }
+      while (keyIter.hasNext()) {
+        Table.KeyValue<String, RepeatedOmKeyInfo> kv = keyIter.next();
+        String key = kv.getKey();
+        RepeatedOmKeyInfo repeatedOmKeyInfo = kv.getValue();
+        // skip the prev key if prev key is present
+        if (skipPrevKey && key.equals(prevKeyPrefix)) {
+          continue;
+        }
+        updateReplicatedAndUnReplicatedTotal(deletedKeyAndDirInsightInfo,
+            repeatedOmKeyInfo);
+        repeatedOmKeyInfoList.add(repeatedOmKeyInfo);
+        if ((repeatedOmKeyInfoList.size()) == limit) {
+          break;
+        }
+      }
+    } catch (IOException ex) {
+      throw new WebApplicationException(ex,
+          Response.Status.INTERNAL_SERVER_ERROR);
+    } catch (IllegalArgumentException e) {
+      throw new WebApplicationException(e, Response.Status.BAD_REQUEST);
+    } catch (Exception ex) {
+      throw new WebApplicationException(ex,
+          Response.Status.INTERNAL_SERVER_ERROR);
+    }
+    return deletedKeyAndDirInsightInfo;
+  }
+
+  private void updateReplicatedAndUnReplicatedTotal(
+      KeyInsightInfoResp deletedKeyAndDirInsightInfo,
+      RepeatedOmKeyInfo repeatedOmKeyInfo) {
+    repeatedOmKeyInfo.getOmKeyInfoList().forEach(omKeyInfo -> {
+      deletedKeyAndDirInsightInfo.setUnreplicatedTotal(
+          deletedKeyAndDirInsightInfo.getUnreplicatedTotal() +
+              omKeyInfo.getDataSize());
+      deletedKeyAndDirInsightInfo.setReplicatedTotal(
+          deletedKeyAndDirInsightInfo.getReplicatedTotal() +
+              omKeyInfo.getReplicatedSize());
+    });
+  }
+
+  /** This method retrieves set of keys/files/dirs which are mapped to
+   * containers in DELETED state in SCM. */
+  @GET
+  @Path("deletedcontainerkeys")
+  public Response getDeletedContainerKeysInfo(
+      @DefaultValue(DEFAULT_FETCH_COUNT) @QueryParam(RECON_QUERY_LIMIT)
+      int limit,
+      @DefaultValue(StringUtils.EMPTY) @QueryParam(RECON_QUERY_PREVKEY)
+      String prevKeyPrefix) {
+    List<KeysResponse> keysResponseList = new ArrayList<>();
+    try {
+      Map<Long, ContainerMetadata> omContainers =
+          reconContainerMetadataManager.getContainers(-1, 0);
+      List<ContainerInfo> deletedStateSCMContainers =
+          containerManager.getContainers(HddsProtos.LifeCycleState.DELETED);
```

Comment:
The code does not check if the `containerManager` are null before calling methods on them, which could result in a NullPointerException if they are null.

Reply:
This is not needed to be checked for an extra null check because this is guice framework injection mechanism and used as standard in whole ozone codebase.